cpp/leaky_core.h

```diff
@@ -57,11 +57,24 @@ class TermState {
 class SignalState {
  public:
   SignalState()
-      : traps(NewDict<Str*, builtin_trap::_TrapHandler*>()),
+      : hooks(NewDict<Str*, builtin_trap::_TrapHandler*>()),
+        traps(NewDict<int, builtin_trap::_TrapHandler*>()),
         nodes_to_run(NewList<syntax_asdl::command_t*>()) {
   }
   void InitShell() {
   }
+  int GetLastSignal() {
+    return last_sig_num;
+  }
+  builtin_trap::_TrapHandler* GetHook(Str* hook_name) {
+    return hooks->get(hook_name);
+  }
+  void AddUserHook(Str* hook_name, builtin_trap::_TrapHandler* handler) {
+    hooks->set(hook_name, handler);
+  }
+  void RemoveUserHook(Str* hook_name) {
+    hooks->remove(hook_name);
+  }
```

Comment:
These will be replaced by translated code in the next PR. Just here to keep the build and tests happy in the meantime